data_loader.py

```diff
@@ -191,7 +191,36 @@ def check_files(self, file_list):
                     return False
             return True
 
-    def validate_files(self, cheat_mode, file_list, max_violations, temp_folder, verbose):
+    def validate_delete_files(self, file_list):
+        validation_result = True
+        with self.driver.session() as session:
+            for txt in file_list:
+                file_encoding = check_encoding(txt)
+                with open(txt, encoding=file_encoding) as in_file:
+                    reader = csv.DictReader(in_file, delimiter='\t')
+                    for org_obj in reader:
+                        obj = self.cleanup_node(org_obj)
+                        id_field = self.schema.get_id_field(obj)
+                        if id_field not in obj.keys():
+                            self.log.error(f'Required id field {id_field} is missing, validation failed')
+                            return False
+                        elif obj[id_field] is None:
+                            self.log.error(f'Required id field {id_field} is None, validation failed')
+                            return False
+                        if NODE_TYPE not in obj.keys():
+                            self.log.error(f'Required node type field {NODE_TYPE} is missing, validation failed')
+                            return True
+                        elif obj[NODE_TYPE] is None:
```

Comment:
Will it handle empty string in type column?

Reply:
The script will return 
**ERROR: (ICDC Schema) get_id_field: "type" field is empty
ERROR: (ICDC Schema) get_id_field: "type" field is empty
ERROR: (Data Loader) Invalid data at line 3: "Node type: "" not found in data model"!**
But the script will not store the "type" missing error in the data validation result file